### PR TITLE
Adding unit tests

### PR DESCRIPTION
**Pull request by fabi200123**
- Adding unit tests for the provider

### CODE REVIEW

**Comment by gabriel-samfira** [comment]
Could you also add a github workflow to run these tests?

**Comment by fabi200123** [comment]
Sure thing! Adding one now!

File: .github/workflows/go-tests.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+      - name: Setup Golang
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: go.mod
+
+      - run: go version
+
+      - name: Run GARM Go Tests
+        run: make go-test
```
Comment (gabriel-samfira) [inline .github/workflows/go-tests.yml:27]: Don't forget the makefile. Or you could just call `go test`

Comment (fabi200123) [reply .github/workflows/go-tests.yml:27]: Added the Makefile now

**Comment by gabriel-samfira** [comment]
Let me know when this is no longer a WiP.

**Pull request merged**